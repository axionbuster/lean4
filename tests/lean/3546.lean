/-!
Issue: "null characters break `readFile`/`getLine`"

When null characters are read from a file using `readFile`, the rest of the file is unexpectedly
cut off, instead of being read as a UTF-8 string (containing null characters).

When hex-dumped, traces of the rest of the file (e.g., the 'b' in "bar"; or corruptions, such as
0x0a) can sometimes be seen.
-/

def test : IO Unit := do
  let tmpFile := "3546.tmp"
  let s := "foo\u0000bar"
  IO.FS.writeFile tmpFile s
  let b := ← IO.FS.readBinFile tmpFile
  IO.println s!"s.length == {s.length}; b.size == {b.size}"
  let u := ← IO.FS.readFile tmpFile
  let v := if s == u then "good" else "bad"
  IO.println v

#eval test
