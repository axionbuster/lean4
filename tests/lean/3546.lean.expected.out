s.length == 7; b.size == 3
good
